src/full/Agda/TypeChecking/Monad/Constraints.hs

```diff
@@ -39,7 +39,7 @@ solvingProblems pids m = verboseBracket "tc.constr.solve" 50 ("working on proble
 isProblemSolved :: (MonadTCEnv m, ReadTCState m) => ProblemId -> m Bool
 isProblemSolved pid =
   and2M (not . Set.member pid <$> asksTC envActiveProblems)
-        (all (not . Set.member pid . constraintProblems) <$> getAllConstraints)
+        ((not . any (Set.member pid . constraintProblems)) <$> getAllConstraints)
```

Comment:
Symmetry broken: original code shows more clearly that this is a big conjunction.  What should be rather done here is to use `Set.notMember`.